parallaxScrolling-1.0.js

```diff
@@ -1,153 +1,166 @@
+        'd': d,
+        'e': e,
+        'g': g
+      };
+
+      // Create the background image
+      var backgroundImg = new Image();
+      backgroundImg.src = src;
+
+      // Once it's loaded
+      backgroundImg.onload = function() {
+        // Set width and height
+        var backgroundWidth = backgroundImg.width;
+        var backgroundHeight = backgroundImg.height;
+
+        // Apply image to DOM element
+        state.el.style.backgroundImage = "url('"+src+"')";
+
+        // Add variables to state
+        state.backgroundWidth = backgroundWidth;
+        state.backgroundHeight = backgroundHeight;
+
+        // Add variables to params
+        params = {
+          'id': id,
+          'scrollDir': scrollDir,
+          'maxOffset': maxOffset
+        };
+
+        // Trigger the resize to set all state vars (resize also calls scroll)
+        return self.resize();
+      }
+
+      initialised = true;
+    };
+
+
+
+    this.scroll = function() {
+      // If the images aren't loaded yet, the args var won't be set so return false
+      if (Object.keys(params).length === 0) {
+        return false;
+      } else {
+        // Setup state variables
+        var scrollTop = state.w.pageYOffset || state.e.scrollTop;
+        var scrollBottom = state.scrollTop + state.windowHeight;
+
+        // Add variables to state
+        state.scrollTop = scrollTop;
+        state.scrollBottom = scrollBottom;
+
+        // Calculate the VERTICAL background position
+        var yPos = 0;
+        if (state.scrollBottom >= state.containerTop && state.scrollTop <= state.containerBottom && (params.maxOffset != 0 || params.maxOffset === 'auto')) {
+          var percentageOfTotalScroll = 100 - (((state.containerBottom - state.scrollTop) / state.containerBottom) * 100);
+
+          if (params.scrollDir === 'up') {
+            percentageOfTotalScroll = 100 - percentageOfTotalScroll;
+          }
+          var totalOffset = state.containerHeight - state.backgroundHeight;
+          if (params.maxOffset != 'auto' && totalOffset < params.maxOffset) {
+            totalOffset = params.maxOffset;
+          }
+
+          yPos = totalOffset * (percentageOfTotalScroll / 100);
+        }
+
+        return yPos;
+      }
+    };
+
+
+
+    this.resize = function() {
+      // If the images aren't loaded yet, the args var won't be set so return false
+      if (Object.keys(params).length === 0) {
+        return false;
+      } else {
+        // Setup state variables
+        var windowHeight = state.w.innerHeight || state.e.clientHeight || state.g.clientHeight;
+        var windowWidth = state.w.innerWidth || state.e.clientWidth || state.g.clientWidth;
+        var containerWidth = state.el.offsetWidth;
+        var containerHeight = state.el.offsetHeight;
+        var containerTop = state.el.offsetTop;
+        var containerBottom = containerTop + containerHeight;
+
+        // Add variables to state
+        state.windowHeight = windowHeight;

```

Comment:
``` state.windowHeight =  state.w.innerHeight || state.e.clientHeight || state.g.clientHeight ```
Do this for all 6 assignments: no need to assign to variables then separately assign to state, just assign to state directly.